reVX/least_cost_xmission/README.md

```diff
@@ -1,25 +1,145 @@
+* *Friction* - Friction indicates areas that are undesirable for transmission. Paths that must cross a friction area will take the shortest possible path. Friction does not affect the cost of a line that runs across it; it merely discourages a path from crossing a friction area, possibly taking a longer and more expensive path.
+* *Barriers* - Barriers are effectively a much stronger deterrent for transmission than friction. Routing will preferentially take a more expensive, and higher friction route before crossing a barrier area. Barriers do not affect the cost of lines that are forced to cross them.
+
+Note that the *friction* and *barriers* layers must be combined together, using a multiplier for barriers, before an analysis is ran.
+
+Layers are created by passing a JSON configuration file (config file) to the [`transmission-layer-creator`](transmission_layer_creator_cli.py) command-line tool. The format of the JSON file is defined using [Pydantic](https://docs.pydantic.dev/latest/) in the `LayerCreationConfig` class of [`transmission_layer_creation.py`](../config/transmission_layer_creation.py). The config file consists of key-value pairs describing necessary files and layer creation options and can trigger a number of different operations depending on it's contents. These operations include:
+
+* Creating a new H5 file to store layers in
+* Creating wet, dry, and landfall masks from a vector
+* Creating wet and dry (TODO) cost layers
+* Combining wet and dry costs
+* Creating frictions and/or barrier layers
+* Combining friction and barrier layers together
+
+## Config File Keys
+### Required Keys
+The following keys are required to run any type of layer creation:
+
+* `template_raster_fpath` - Template GeoTIFF with the shape, CRS, and transform to use for the project. Data in the raster is ignored.
+* `h5_fpath` - H5 file to store layers in. `h5_fpath` will be created if it does not exist and `existing_h5_fpath` is set.
+
+### Optional Keys
+Most of these keys are optional and affect how the layer creation runs. Some, like `land_mask_vector_fname`, may need to be specified once to generate the mask files, and then can be disabled.
+
+* `existing_h5_fpath` - Existing reV H5 file with the same shape as `template_raster_fpath`. Metadata is copied from `existing_h5_fpath` to `h5_fpath`. This must be defined to create `h5_fpath` if it does not exist.
+* `land_mask_vector_fname` - Polygon GeoPackage or shapefile that defines what areas are considered wet and dry. Features in the file are assumed to represent dry land. The boundaries of features are used to determine the landfall cells.
```

Comment:
I'm concern that some of these options are a bit confusing for the users, e.g. land_mask_vector_fname. That only needs to be specified once. The code loads it, builds the masks, and saves them as tiffs. Everytime after, the code can just load the tiffs which is faster. Ideally the code would check the modification date on the vector file and rebuild the rasters if it's newer than the raster. OR, if the user changed the vector name, though I'm not quite sure how to track that. I might also be going a little too far down the rabbit hole here.

Reply:
Well, could we pull this out into it's own utility function (and maybe CLI command?) that the user would run before computing costs? I think it's ok for the code to simply require a mask TIFF input, especially if we provide them utility elsewhere to create it. Am I on the right track here or is that not what you were asking?

Reply:
That's a great idea, and it would be easy enough to have a `build-masks` argument for the same CLI tool. Probably could do the same for creating the new H5 file.